projects/common/src/time/time-range.service.ts

```diff
@@ -62,16 +66,44 @@ export class TimeRangeService {
     this.setTimeRange(this.getCurrentTimeRange());
   }
 
+  private globalTimeRangeInitialization(): Observable<TimeRange> {
+    return this.navigationService.navigation$.pipe(
+      take(1), // Wait for first navigation
+      switchMap(activatedRoute => activatedRoute.queryParamMap), // Get the params from it
+      take(1), // Only the first set of params
+      map(paramMap => paramMap.get(TimeRangeService.TIME_RANGE_QUERY_PARAM)), // Extract the time range value from it
+      filter((timeRangeString): timeRangeString is string => !isEmpty(timeRangeString)), // Only valid time ranges
+      map(timeRangeString => this.timeRangeFromUrlString(timeRangeString)),
+      catchError(() => EMPTY)
+    );
+  }
+
+  private pageTimeRangeInitialization(): Observable<TimeRange> {
+    return this.navigationService.navigation$.pipe(
+      filter(
+        activatedRoute => !isNil(activatedRoute.snapshot.queryParamMap.get(TimeRangeService.TIME_RANGE_QUERY_PARAM))
+      ),
+      map(activeRoute => {
+        const timeRangeQueryParamString = activeRoute.snapshot.queryParamMap.get(
+          TimeRangeService.TIME_RANGE_QUERY_PARAM
+        );
+
+        return this.timeRangeFromUrlString(timeRangeQueryParamString!);
```

Comment:
So when do we want to emit? 
- Current code here would change the time range every time we navigate. That would break flows like drilling down in the same TR
- We could do it only when the TR string changes (e.g. 1h -> 1d), but that means if we change top level items when the setting is the same, we wouldn't refresh (this is the behavior today, so seems acceptable)
- If we need to know when to refresh and when not to, we'll likely need another param

Reply:
No this only emits when we  navigate and have a time query param. Drill behaviour still works as expected.